agent/core/src/main/java/org/jolokia/restrictor/RestrictorFactory.java

```diff
@@ -115,9 +142,35 @@ public static PolicyRestrictor lookupPolicyRestrictor(String pLocation) throws I
                 is = RestrictorFactory.class.getResourceAsStream(path);
             }
         } else {
-            URL url = new URL(pLocation);
-            is = url.openStream();
+            URLConnectionResult urlConnectionResult = buildUrlConnection(pLocation);
+            is = urlConnectionResult.urlConnection.getInputStream();
         }
         return is != null ? new PolicyRestrictor(is) : null;
     }
+
+    /* default */ static URLConnectionResult buildUrlConnection(String pLocation)
+            throws MalformedURLException, IOException {
+        Objects.requireNonNull(pLocation);
+
+        Matcher matcher = URL_PATTERN.matcher(pLocation);
+
+        String url = pLocation;
+        String basicAuthHeaderValue = null;
+
+        // We got a basic-auth user and password
+        if (matcher.matches() && matcher.group(3) != null && matcher.group(4) != null) {
+            url = matcher.group(1) + matcher.group(5);
```

Comment:
See above, you could also used named groups which often make things easier to understand

```java
url = matcher.group("protocol") + matcher.group("rest");
```